samples/matter/common/src/dfu_over_smp.cpp

```diff
@@ -35,12 +35,21 @@ void DFUOverSMP::Init(DFUOverSMPRestartAdvertisingHandler startAdvertisingCb)
 	img_mgmt_set_upload_cb(UploadConfirmHandler);
 
 	memset(&mBleConnCallbacks, 0, sizeof(mBleConnCallbacks));
-	mBleConnCallbacks.connected = OnBleConnect;
+	mBleConnCallbacks.connected = nullptr;
 	mBleConnCallbacks.disconnected = OnBleDisconnect;
 	bt_conn_cb_register(&mBleConnCallbacks);
-
-	k_work_init(&mFlashSleepWork, [](k_work *) { GetFlashHandler().DoAction(FlashHandler::Action::SLEEP); });
-	k_work_init(&mFlashWakeUpWork, [](k_work *) { GetFlashHandler().DoAction(FlashHandler::Action::WAKE_UP); });
+	mgmt_register_evt_cb([](uint8_t opcode, uint16_t group, uint8_t id, void *arg) {
+		switch (opcode) {
+		case MGMT_EVT_OP_CMD_RECV:
+			GetFlashHandler().DoAction(FlashHandler::Action::WAKE_UP);
+			break;
+		case MGMT_EVT_OP_CMD_DONE:
```

Comment:
I am wondering if there is any possibility that the dfu is abandoned for whatever reason and the handler has no chance to execute at all resulting in this callback not being triggered as well.

Reply:
If I read the code correctly the callback should be called even if the handler returns an error, so we should be safe, right?

Reply:
That is, we enable and disable the flash for each command, not once per DFU.

Reply:
Error path also calls the event.

> That is, we enable and disable the flash for each command, not once per DFU.

Yes, this is more aggressive PM approach, but it is still less aggressive than doing it directly in flash_map operations.
Depends what you want to achieve; I think this is good approach because if you would do that once per DFU you would have to handle situation when DFU gets interrupted for longer time: in such case a flash would be left powered on unless you have some timeout to power it down.
In this solution you enable the flash when you receive an request and even if comm breaks  the code for powering it down will be executed.